pkg/agent/controller/networkpolicy/cache.go

```diff
@@ -911,3 +906,35 @@ func (c *ruleCache) unionAppliedToGroups(groupNames []string) (v1beta.GroupMembe
 	}
 	return set, anyExists
 }
+
+// processServiceGroupIDUpdate gets names of AppliedToGroup by Service NamespacedName.
+func (c *ruleCache) processServiceGroupIDUpdate(svcStr string) {
+	c.appliedToSetLock.RLock()
+	defer c.appliedToSetLock.RUnlock()
+
+	// Reprocess rules if the Service referred by this rule's ToServices has updated.
+	toSvcRules, err := c.rules.ByIndex(toServicesIndex, svcStr)
+	if err != nil {
+		return
+	}
+	for _, toSvcRule := range toSvcRules {
+		c.dirtyRuleHandler(toSvcRule.(*rule).ID)
+	}
+
+	// Reprocess rules if the Service referred by rule's AppliedToGroup has updated.
+	strListSvcRef := strings.Split(svcStr, "/")
+	if len(strListSvcRef) != 2 {
+		return
+	}
+	member := &v1beta.GroupMember{
+		Service: &v1beta.ServiceReference{
+			Name:      strListSvcRef[1],
+			Namespace: strListSvcRef[0],
+		},
+	}
+	for group, memberSet := range c.appliedToSetByGroup {
+		if memberSet.Has(member) {
```

Comment:
I'm thinking what if a group has a Pod which has same name as the service, looks like it will be matched by accident, should we have add different prefixes for different kinds of references when hashing a member?

Reply:
Good catch! Added.